src/Output/Output_BasePowerSpectrum.cpp

```diff
@@ -58,6 +58,11 @@ void Output_BasePowerSpectrum( const char *FileName, const long TVar )
 #  endif // #  if ( SUPPORT_FFTW == FFTW3 ) ... # else
 #  endif // #  ifdef SERIAL ... # else
 
+// check integer overflow (in a non-rigorous way)
+   if ( total_local_size <= 0 )
```

Comment:
For which platforms and compilers does this work? According to https://users.cs.utah.edu/~regehr/papers/overflow12.pdf the overflow behaviour for signed integers should be undefined in C. 
For unsigned types, the behaviour seems to be to wrap around. Could we make mpi_index_int an unsigned integer for FFTW2 and then check whether `(long) local_nx*local_ny*local_nz > mpi_index_int`to detect an overflow more reliably?

Reply:
Based on [this](https://stackoverflow.com/questions/3944505/detecting-signed-overflow-in-c-c) link, I suspect your solution might still not be safe as behaviour after an overflow is undefined according to the C standard. They suggest using a type that is large enough to hold the result and then checking against INT_MIN and INT_MAX. I believe this solution should be robust. 

```
assert(LLONG_MAX>INT_MAX);
long long size_long = (long long)local_nx*(long long)local_ny*(long long)local_nz;
if (size_long < INT_MIN || size_long > INT_MAX) 
    Aux_Error( ERROR_INFO, "size_long (%ld) != size_int (%ld) and/or total_local_size (%ld) <= 0 for FFT\n"
                   "        --> Likely due to integer overflow. Consider using more MPI processes !!\n",
               size_long, (long)size_int, (long)total_local_size );
```